miden/tests/integration/operations/stack_ops.rs

```diff
@@ -28,3 +28,407 @@ fn swapdw() {
+        let mut expected_values = test_values[..8].to_vec();
+        let mut b = test_values[8..].to_vec();
+        expected_values.reverse();
+        b.reverse();
+        expected_values.append(&mut b);
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn drop_proptest(test_values in prop::collection::vec(any::<u64>(), 16)) {
+        let asm_op = "drop";
+        let mut expected_values = test_values.to_vec();
+        expected_values.remove(15);
+        expected_values.reverse();
+        expected_values.push(0);
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn dropw_proptest(test_values in prop::collection::vec(any::<u64>(), 16)) {
+        let asm_op = "dropw";
+        let mut expected_values = test_values.to_vec();
+        expected_values.truncate(12);
+        expected_values.reverse();
+        expected_values.append(&mut vec![0; 4]);
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn padw_proptest(test_values in prop::collection::vec(any::<u64>(), 16)) {
+        let asm_op = "padw";
+        let mut expected_values = test_values.to_vec();
+        expected_values.drain(0..4);
+        expected_values.append(&mut vec![0; 4]);
+        expected_values.reverse();
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn dup_proptest(test_values in prop::collection::vec(any::<u64>(), 16)) {
+        let asm_op = "dup";
+        let mut expected_values = test_values.to_vec();
+        expected_values.remove(0);
+        expected_values.push(expected_values[expected_values.len() - 1]);
+        expected_values.reverse();
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn dupw_proptest(test_values in prop::collection::vec(any::<u64>(), 16)) {
+        let asm_op = "dupw";
+        let mut expected_values = test_values.to_vec();
+        expected_values.drain(0..4);
+        let mut a = test_values[12..].to_vec();
+        expected_values.append(&mut a);
+        expected_values.reverse();
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn swap_proptest(test_values in prop::collection::vec(any::<u64>(), 16)) {
+        let asm_op = "swap";
+        let mut expected_values = test_values.to_vec();
+        expected_values.reverse();
+        expected_values.swap(0, 1);
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn swapw_proptest(test_values in prop::collection::vec(any::<u64>(), 16)) {
+        let asm_op = "swapw";
+        let mut expected_values = test_values.to_vec();
+        let mut a = test_values[12..].to_vec();
+        let mut b = test_values[8..12].to_vec();
+        expected_values.truncate(8);
```

Comment:
nit: you could use `expected_values.split_off` here instead of `to_vec` and `truncate`.
```
        let mut a = expected_values.split_off(12);
        let mut b = expected_values.split_off(8);
```